staging/src/k8s.io/client-go/examples/unstructured-parsing/main_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unstructuredparsing
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/runtime/serializer/streaming"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+)
+
+// Example yaml input
+var in = `
+
+# Comments will be removed during parsing
+kind: Foo
+apiVersion: v1alpha1
+spec:
+  foo: 1
+metadata:
+  name: foo
+
+---
+
+kind: Bar
+apiVersion: v1alpha1
+spec:
+    bar: 2
+
+metadata:
+  name: bar
+
+`
+
+// The expected output - note the normalization of spacing & field order
+var expected = `---
+apiVersion: v1alpha1
+kind: Foo
+metadata:
+  name: foo
+spec:
+  foo: 1
+---
+apiVersion: v1alpha1
+kind: Bar
+metadata:
+  name: bar
+spec:
+  bar: 2
+`
+
+// TestUnstructuedParsing demonstrates how to parse a yaml file, using the unstructured codecs.
+func TestUnstructuredParsing(t *testing.T) {
+	// Build a yaml decoder with the unstructured Scheme
+	yamlDecoder := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
```

Comment:
I wasn't sure if there was a way to auto-detect json vs yaml and whether I actually should be using a different API entirely